### DISCUSSION TITLE
Setting variable in Jupyter Notebook

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1165] [Q&A] [answered]
Trying 

```
%%bash
gff="/Users/sr320/Downloads/ROSLIN/ncbi_dataset/data/GCF_902806645.1/genomic.sorted.gff"
```

but getting nothing with

```
!echo $gff
```

**Comment by laurahspencer** [comment]
Try 
! echo ${gff}

**Comment by kubu4** [comment]
Is that echo statement being run in the same cell? If not, you have to set an environmental variable so that the variable can be used throughout entire notebook. See one of my recent notebooks as an example.

**Comment by yaaminiv** [comment, marked as answer]
solution:

set variable without `%%bash`

**Comment by kubu4** [comment]
Haven't we experienced weird failures in the past with @yaaminiv's solution? I feel that's why I began declaring environmental variables (which is kind of annoying), instead of that approach.

**Reply by yaaminiv** [reply]
I ran into issues using variables in loops (ex. using bedtools)! But outside of that, haven't had issues. Guess it depends on what you need to do with the variable

**Comment by kubu4** [comment]
So strange! I wish we had some idea of what happens with that.
